### PR TITLE
Reenable media feature range notation Stylelint check

### PR DESCRIPTION
**Pull request by basil**
A while back I had disabled the (default) `media-feature-range-notation` Stylelint check because it was not supported by `postcss-less` or HtmlUnit. We are now off of `postcss-less`, so HtmlUnit is the only thing standing in our way. Since it doesn't seem sustainable to be stuck on old CSS features forever, this PR introduces `postcss-preset-env` to handle compiling newer CSS into older idioms for older browsers much the same way as Babel (which we already use) does this for JavaScript. `postcss-preset-env` is a supserset of `autoprefixer`, so I replaced any references to `autoprefixer` with references to `postcss-preset-env`. Rather than enable the default set of transformations, which made a lot of changes to the generated CSS, I enabled just the `media-query-ranges` transformation. Before enabling `media-query-ranges`, the Stylelint check was passing but the generated CSS contained the modifications in this PR and the new test from #8148 was failing in HtmlUnit. After enabling `media-query-ranges`, the Stylelint check was still passing, but the generated CSS was the same as before and the new test from #8148 started passing in HtmlUnit. What this means is that as we want to use new CSS features (or as Stylelint guides us to those new features) that are not supported by HtmlUnit, we can do so and rely on this framework to compile the generated CSS to the lowest common denominator that will work in HtmlUnit.

### Testing done

In debug mode, verified that only `autoprefixer` and `media-query-ranges` were enabled and that the generated CSS was the same as before. Verified that Stylelint passed with the new check enabled (and failed when I intentionally introduced a violation). Verified that the new test from #8148 passed.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8149"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
